htdocs/core/lib/functions.lib.php

```diff
@@ -38,6 +38,59 @@
 include_once DOL_DOCUMENT_ROOT .'/core/lib/json.lib.php';
 
 /**
+ * Function to transform field from a dictionnary
+ * 
+ * @param 	$db	Database 	Object
+ * @param	$string			String to transform
+ * @param	$tableName		Table of record
+ * @param	$fieldName		Field name
+ */
+function dol_transform_field(&$db, &$string, $tableName, $fieldName) {
+	if(dolibarr_get_const($db, 'DICT_FIELD_CASES')) {
+		$param = dolibarr_find_dictionnary($db, 'c_field_cases', array('table_name'=>$tableName, 'field_name'=>$fieldName));
+		if($param) {
+			switch(strtoupper($param->transformation)) {
+				case 'UPPERCASE':
+					$string = strtoupper($string);
+					break;
+				case 'UCWORDS':
+					dol_ucwords($string);
+					break;
+				default:
+					if(preg_match($param->transformation, null) !== false) {
+						$string = preg_replace(str_replace('/[', '/[^', $param->transformation),'',$string);
+					}
+					break;
+			}
+		}
+	}
+}
+
+/**
+ * UCWords with defined separators
+ * (e.g.: Chateau De L'atour-maubourg --> Château De L'Atour-Maubourg)
```

Comment:
English example would be better.
On a funny note, I doubt that the code will add the circumflex accent ;)